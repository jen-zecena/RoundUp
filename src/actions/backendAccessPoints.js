
export const BACKEND_HOSTNAME = 'http://localhost:8080/';
/*
  Auth
*/
export const USERS_URL = BACKEND_HOSTNAME + 'api/users/'
export const USERS_LOGIN_URL = USERS_URL + 'login/'


/*
  Events
*/

/*
  Rsvps
*/

/*
  Subscriptions
*/
