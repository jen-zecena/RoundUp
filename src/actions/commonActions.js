import { API_REQUEST } from './types/commonActionTypes';

export const apiRequest = () => ({
  type: API_REQUEST
})
