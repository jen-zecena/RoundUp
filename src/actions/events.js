import axios from 'axios';
import { stopSubmit } from 'redux-form';
import { reset } from 'redux-form';

import {
  ADD_EVENT,
  GET_EVENT,
  UPDATE_EVENT,
  DELETE_EVENT,
  GET_EVENTS
} from './types/events';

import history from '../history';
import { tokenConfig } from './auth';

// actions related to the events **
/**
  @param userId: the ID of the user who is creating the event
  @param description: the description of the event to be created
  @param eventTime: the time the event will take place
  @param poster: the url for the poster corresponding to this event
  @param name: the name of the event given by the user
  @param location: the location of the event as given by the user

  This method adds a new event by passing new event information to the backend to be added
  to the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the added event.
*/
export default addEvent = (userId, description, eventTime, poster, name, location, tags) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to add the new event.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  try {
    dispatch({
      type: ,
      payload: response.data
    });
  } catch (error) {
    dispatch({
      type:
    })
  }
}

/**
  @param eventId: the id of the specific event

  This method retrieves event information from the database by making
  a request to the backend.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved event.
*/
export default getEvent = (eventId) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve the event information.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  try {
    dispatch({
      type: ,
      payload: response.data
    });
  } catch (error) {
    dispatch({
      type:
    })
  }
}

/**
  @param eventId:  the id of the event to be updated
  @param userId: the user id of the user trying to update the event
  @param description: the new description for the event
  @param eventTime: the new time for the event
  @param poster: the url of the event poster
  @param name: the new name for the event
  @param location: the new location for the event

  This method modifies an event's information by makin a request to
  the backend to modify the specific event's information in the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the updated event.
*/
export default updateEvent = (eventId, userId, description, eventTime, poster, name, location, tags) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to modify the specified event.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  try {
    dispatch({
      type: ,
      payload: response.data
    });
  } catch (error) {
    dispatch({
      type:
    })
  }
}

/**
  @param eventId: the id of the specific event

  This method deletes an event by making a request to the backend to delete the
  specified event from the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the deleted event.
*/
export default deleteEvent = (eventId) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to delete the event.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  try {
    dispatch({
      type: ,
      payload: response.data
    });
  } catch (error) {
    dispatch({
      type:
    })
  }
}

/**
  @param tags: a list of tags associated with the event
  @param status: the status of events user is interested in

  This method retrieves events by making a request ot the backend to extract events with
  specified parameters.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved events.
*/
export default getEvents = (tags, status) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve specified events.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  try {
    dispatch({
      type: ,
      payload: response.data
    });
  } catch (error) {
    dispatch({
      type:
    })
  }
}
