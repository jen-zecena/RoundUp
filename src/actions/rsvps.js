import axios from 'axios';
import { stopSubmit } from 'redux-form';

import {
  ADD_RSVP,
  DELETE_RSVP,
  GET_USER_RSVPS,
  GET_EVENT_RSVPS,
  GET_ATTENDEES
} from './types/rsvps';

// ** rsvp **
/**
  @param userID: the id of the user who is rsvp'ing to the event
  @param eventID: the id of the event to be rsvp'ing to

  This method adds a new rsvp by passing new rsvp information to be
  added to the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the added rsvp.
*/
export const addRsvp = ({ userID, eventID }) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to add the new rsvp.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  const response = await axios.post(`/rsvps`, { userID, eventID }, tokenConfig(getState));
  dispatch({
    type: ADD_RSVP,
    payload: response.data
  });
}


/**
  @param eventID: the id of the event to be rsvp'ing to

  This method retrieves a list of attendees for an event.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved attendees.
*/
export const getAttendees = (eventID) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve the list of attendees.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  const response = await axios.get(`/events/${eventID}/attendees`, tokenConfig(getState));
  dispatch({
    type:GET_ATTENDEES,
    payload: response.data
  });
}

/**
  @param userID: the id of the user in question

  This method retrieves a user's rsvp history.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved rsvps.
*/
export const getUserRsvps = (email, status) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve rsvp history.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  const response = await axios.get(`/users/${email}/rsvps/${status}`, tokenConfig(getState));
  dispatch({
    type: GET_USER_RSVPS,
    payload: response.data
  });
}

/**
  @param eventID: the id of the event in question

  This method retrieves the list of an event's rsvps.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved rsvps.
*/
export const getEventRsvps = (eventID) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve an event's rsvps.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  const response = await axios.get(`/events/${eventID}/rsvps`, tokenConfig(getState));
  dispatch({
    type: GET_EVENT_RSVPS,
    payload: response.data
  });
}

/**
  @param userID: the id of the user who is rsvp'ing to the event
  @param eventID: the id of the event to be rsvp'ing to

  This method deletes an rsvp by making a request to the backend to delete it
  from the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the deleted rsvp.
*/
export const deleteRsvp = ({ userID, eventID }) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to delete the specified rsvp object.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  const response = await axios.delete(`/rsvps`, { userID, eventID }, tokenConfig(getState));
  dispatch({
    type: DELETE_RSVP,
    payload: response.data
  });
}
