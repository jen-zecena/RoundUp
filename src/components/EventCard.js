import React from "react";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import { Link } from "react-router-dom";
import { BrowserRouter as Router} from "react-router-dom";


/**
  @param pops: the properties of the event that the EventCard will render
  @return: html code with embeded javascript that references properties in the prop passed into this function
  It will include the structure and content of a Event Poster Card

  The react component that this function returns will be populated with different data depending on the props
  passed into it. This component is very modular and we will display many of them in our final events display page

**/

function EventCard(props) {
  return (
    <Router>
    <Card style={{ width: "18rem" }} className="card">
      <Card.Img variant="top" src={props.posterUrl} />
      <Card.Body>
        <Card.Title className="tile"> {props.name} </Card.Title>
        <Card.Subtitle className="mb-2 text-muted">
          {props.date} {props.time} {props.location}
        </Card.Subtitle>
        <Card.Text className="text">{props.description}</Card.Text>
        <Link to={props.eventLink}>
          <Button variant="danger" className="button">
            {" "} More Information {" "}
          </Button>
        </Link>
      </Card.Body>
    </Card>
    </Router>
  );
}

export default EventCard;
