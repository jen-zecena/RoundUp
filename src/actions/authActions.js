import axios from 'axios';
import { stopSubmit } from 'redux-form';
import { apiRequest } from './commonActions';

import {
  LOADING_USER,
  LOADING_USER_SUCCESS,
  LOADING_USER_FAIL,
  REGISTRATION_SUCCESS,
  REGISTRATION_FAIL,
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT_SUCCESS,
  LOGOUT_FAIL
} from './types/authActionTypes';

import store from '../store';

export const loadUserAction = () => async (dispatch, getState) => {
  /*
    1. tell the web application that the user information is being loaded.
    2. retrieve authentication token from local storage
    3. use authentication token to rtrieve user information from database
    4. if the user is successfully loaded, tell authentication reducer about
    success and pass along retrieved user information
    5. otherwise, tell the authentication reducer about the failed attempt.

  */
  dispatch({
    type: LOADING_USER
  });

  dispatch(apiRequest());
  try {
    const response = await axios.get('/users/', tokenConfig(getState));
    dispatch({
      type: LOADING_USER_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: LOADING_USER_FAIL,
      error: error.message
    });
    return error;
  }
}

/**
  @param email: the email of the user
  @param password: the password of the user

  This method logs in the user by making a request to the backend to
  check for the existence of a user with the provided email and password.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the user's state.
*/
export const loginUserAction = (email, password)  => async dispatch => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to login the user.
    3. If the login was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */

  // creating request headers and body
  const config = getHeaders();
  const body = JSON.stringify({ email, password });

  // get token with this method
  // btoa("email:password")


  dispatch(apiRequest());
  try {
    const response = await axios.post('/users/login', body, config);
    dispatch({
      type: LOGIN_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: LOGIN_FAIL,
      error: error.message
    });
    // dispatch(stopSubmit('loginForm', error.response.data));
    return error;
  }
}


/**
  @param firstName: the first name of the new user
  @param lastName: the last name of the new user
  @param email: the email of the new user
  @param password: the password of the new user
  @param campus: the campus of the new user
  @param cohort: the cohort of the new user

  This method registers the user by passing user information to the backend to be added
  to the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the user's state.
  */
export const registerUserAction = (firstName, lastName, email, password, campus) =>  async dispatch => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to register the user.
    3. If the registration was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  // creating request headers and body
  const config = getHeaders();
  const body = JSON.stringify({ firstName, lastName, email, password, campus });

  dispatch(apiRequest());
  try {
    const response = await axios.post('/users/register', body,  config)
    dispatch({
      type: REGISTRATION_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: REGISTRATION_FAIL,
      error: error.message
    });
    // dispatch(stopSubmit('registrationForm', error.response.data));
    return error;
  }
}


/**
  This method logs out the user by passing user information to the backend.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the user's authentication state
*/
export const logoutUserAction = () =>  async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to logout the user.
    3. If the logout was successful, tell authentication reducer about
    success.
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  const token = getToken(getState);
  if (token) {
    dispatch({
      type: LOGOUT_SUCCESS
    })
  } else {
    dispatch({
      type: LOGOUT_FAIL
    })
  }
}

/**
  @param getState: the state
  This method is a helper function to retrieve the authentication token
*/
export const tokenConfig = getState => {
  // Get token
  const token = getToken(getState);

  // Headers
  const config = {
    headers: {
      'Content-Type': 'application/json'
    }
  };

  if (token) {
    config.headers['Authorization'] = `Basic ${token}`;
  }

  return config;
};

function getToken(getState) {
  var token = null;
  try {
    token = getState().auth.token;
  } catch (error) {}
  return token;
}

function getHeaders() {
  return {
    headers: {
      'Content-Type': 'application/json',
    }
  };
}
