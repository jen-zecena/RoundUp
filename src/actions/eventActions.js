import axios from 'axios';
import { stopSubmit } from 'redux-form';
import { reset } from 'redux-form';
import { apiRequest } from './commonActions';

import {
  ADD_EVENT_SUCCESS,
  ADD_EVENT_FAIL,
  GET_EVENT_SUCCESS,
  GET_EVENT_FAIL,
  UPDATE_EVENT_SUCCESS,
  UPDATE_EVENT_FAIL,
  DELETE_EVENT_SUCCESS,
  DELETE_EVENT_FAIL,
  // GET_EVENTS_SUCCESS,
  // GET_EVENTS_FAIL,
  GET_EVENTS_BY_TAGS_SUCCESS,
  GET_EVENTS_BY_TAGS_FAIL,
  GET_EVENTS_BY_NAME_SUCCESS,
  GET_EVENTS_BY_NAME_FAIL,
  GET_EVENTS_BY_OWNER_SUCCESS,
  GET_EVENTS_BY_OWNER_FAIL,
  GET_EVENTS_BY_TIME_SUCCESS,
  GET_EVENTS_BY_TIME_FAIL,
  GET_EVENT_ATTENDEES_SUCCESS,
  GET_EVENT_ATTENDEES_FAIL
} from './types/eventActionTypes';

import {
  EVENTS_URL
} from './backendAccessPoints';

import history from '../history';
import { tokenConfig } from './authActions';

// actions related to the events **
/**
  @param uID: the ID of the user who is creating the event
  @param description: the description of the event to be created
  @param eventTime: the time the event will take place
  @param poster: the url for the poster corresponding to this event
  @param name: the name of the event given by the user
  @param location: the location of the event as given by the user

  This method adds a new event by passing new event information to the backend to be added
  to the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the added event.
*/
export const addEventAction = ({ uID, description, eventTime, poster, name, location, tags }) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to add the new event.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.post(EVENTS_URL, { uID, description, eventTime, poster, name, location, tags }, tokenConfig(getState));
    dispatch({
      type: ADD_EVENT_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: ADD_EVENT_FAIL,
      error: error.message
    });
    return error;
  }
}

/**
  @param eventID: the id of the specific event

  This method retrieves event information from the database by making
  a request to the backend.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved event.
*/
export const getEventAction = (eventID) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve the event information.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    console.log("event Action");
    const response = await axios.get(`${EVENTS_URL}${eventID}/`, tokenConfig(getState));
    dispatch({
      type: GET_EVENT_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: GET_EVENT_FAIL,
      error: error.message
    });
    return error;
  }
}

/**
  @param eventID:  the id of the event to be updated
  @param uID: the user id of the user trying to update the event
  @param description: the new description for the event
  @param eventTime: the new time for the event
  @param poster: the url of the event poster
  @param name: the new name for the event
  @param location: the new location for the event

  This method modifies an event's information by makin a request to
  the backend to modify the specific event's information in the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the updated event.
*/
export const updateEventAction = ({ eventID, uID, description, eventTime, poster, name, location, tags }) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to modify the specified event.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.put(`${EVENTS_URL}${eventID}/`, { eventID, uID, description, eventTime, poster, name, location, tags }, tokenConfig(getState));
    dispatch({
      type: UPDATE_EVENT_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: UPDATE_EVENT_FAIL,
      error: error.message
    });
    return error;
  }

}

/**
  @param eventID: the id of the specific event

  This method deletes an event by making a request to the backend to delete the
  specified event from the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the deleted event.
*/
export const deleteEventAction = (eventID, uID) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to delete the event.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.delete(EVENTS_URL, tokenConfig(getState), { 'eventID': eventID, 'uID': uID });
    dispatch({
      type: DELETE_EVENT_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: DELETE_EVENT_FAIL,
      error: error.message
    });
    return error;
  }
}

// /**
//   @param tags: a list of tags associated with the event
//   @param search: the key word to search events by
//   @param uID: the owner's user id
//   @param status: the status of events user is interested in
//   @param fromTime: the beginning of the events time window
//   @param toTime: the end of the events time window
//
//   This method retrieves eventsthat have particular tags by making a request ot the backend to extract events with
//   specified parameters.
//
//   The method dispatches the result of the action to the reducer, which
//   informs the web app of the retrieved events.
// */
// export const getEventsAction = ({ tags, search, uID, status, fromTime, toTime }) => async (dispatch, getState) => {
//   dispatch(apiRequest());
//   try {
//     const response = await axios.get(EVENTS_URL, {params: { tags: tags, search: search, status: status, fromTime: fromTime, toTime: toTime }});
//     dispatch({
//       type: GET_EVENTS_SUCCESS,
//       payload: response.data.events
//     });
//     return response;
//   } catch (error) {
//     dispatch({
//       type: GET_EVENTS_FAIL,
//       error: error.message
//     });
//     return error;
//   }
// }

/**


  This method retrieves eventsthat have particular tags by making a request ot the backend to extract events with
  specified parameters.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved events.
*/
export const getEventsByTagsAction = (tags, status) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve specified events.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.get(EVENTS_URL + "tags/", {tags, status}, tokenConfig(getState));
    dispatch({
      type: GET_EVENTS_BY_TAGS_SUCCESS,
      payload: response.data.events
    });
    return response;
  } catch (error) {
    dispatch({
      type: GET_EVENTS_BY_TAGS_FAIL,
      error: error.message
    });
    return error;
  }
}


/**
  @param uID: the owner's user id
  @param status: the status of events user is interested in

  This method retrieves events belonging to a particular owner by making a request ot the backend to extract events with
  specified parameters.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved events.
*/
export const getEventsByOwnerAction = (uID, status) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve specified events.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.get(EVENTS_URL + `owner/${uID}/`, {status}, tokenConfig(getState));
    dispatch({
      type: GET_EVENTS_BY_OWNER_SUCCESS,
      payload: response.data.events
    });
    return response;
  } catch (error) {
    dispatch({
      type: GET_EVENTS_BY_OWNER_FAIL,
      error: error.message
    });
    return error;
  }
}


/**
  @param search: the key word to search events by
  @param status: the status of events user is interested in

  This method retrieves events that that a key word by making a request ot the backend to extract events with
  specified parameters.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved events.
*/
export const getEventsByNameAction = (search, status) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve specified events.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.get(EVENTS_URL + "search/", {search, status}, tokenConfig(getState));
    dispatch({
      type: GET_EVENTS_BY_NAME_SUCCESS,
      payload: response.data.events
    });
    return response;
  } catch (error) {
    dispatch({
      type: GET_EVENTS_BY_NAME_FAIL,
      error: error.message
    });
    return error;
  }
}


/**
  @param fromTime: the beginning of the events time window
  @param toTime: the end of the events time window

  This method retrieves events within a time window by making a request ot the backend to extract events with
  specified parameters.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved events.
*/
export const getEventsByTimeAction = (fromTime, toTime) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve specified events.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.get(EVENTS_URL + "timeframe/",{ params: {fromTime, toTime}});
    dispatch({
      type: GET_EVENTS_BY_TIME_SUCCESS,
      payload: response.data.events
    });
    return response;
  } catch (error) {
    dispatch({
      type: GET_EVENTS_BY_TIME_FAIL,
      error: error.message
    });
    return error;
  }
}


/**
  @param eventID: the id of the event to be rsvp'ing to

  This method retrieves a list of attendees for an event.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved attendees.
*/
export const getEventAttendeesAction = (eventID) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve the list of attendees.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.get(`${EVENTS_URL}${eventID}/attendees/`, tokenConfig(getState));
    dispatch({
      type: GET_EVENT_ATTENDEES_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: GET_EVENT_ATTENDEES_FAIL,
      error: error.message
    });
  }
}
