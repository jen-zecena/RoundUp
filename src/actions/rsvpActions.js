import axios from 'axios';
import { stopSubmit } from 'redux-form';
import { apiRequest } from './commonActions';

import { tokenConfig } from './authActions';

import {
  ADD_RSVP_SUCCESS,
  ADD_RSVP_FAIL,
  DELETE_RSVP_SUCCESS,
  DELETE_RSVP_FAIL,
  GET_USER_RSVPS_SUCCESS,
  GET_USER_RSVPS_FAIL,
  // GET_ATTENDEES_SUCCESS,
  // GET_ATTENDEES_FAIL
} from './types/rsvpActionTypes';

import {
  USERS_URL,
  RSVPS_URL
} from './backendAccessPoints'

// ** rsvp **
/**
  @param uID: the id of the user who is rsvp'ing to the event
  @param eventID: the id of the event to be rsvp'ing to

  This method adds a new rsvp by passing new rsvp information to be
  added to the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the added rsvp.
*/
export const addRsvpAction = ({email, name, eID, time}) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to add the new rsvp.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.post(RSVPS_URL, { email, name, eID, time }, tokenConfig(getState));
    dispatch({
      type: ADD_RSVP_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: ADD_RSVP_FAIL,
      error: error.message
    });
  }
}

/**
  @param uID: the id of the user in question

  This method retrieves a user's rsvp history.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the retrieved rsvps.
*/
export const getUserRsvpsAction = (email, status) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to retrieve rsvp history.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.get(`${USERS_URL}${email}/rsvps/${status}`, tokenConfig(getState));
    dispatch({
      type: GET_USER_RSVPS_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: GET_USER_RSVPS_FAIL,
      error: error.message
    });
  }
}

/**
  @param uID: the id of the user who is rsvp'ing to the event
  @param eventID: the id of the event to be rsvp'ing to

  This method deletes an rsvp by making a request to the backend to delete it
  from the database.

  The method dispatches the result of the action to the reducer, which
  informs the web app of the deleted rsvp.
*/
export const deleteRsvpAction = ( email, eventID) => async (dispatch, getState) => {
  /*
    1. Create the request configuration and stringify the parameters
    2. Make a request to the backend to delete the specified rsvp object.
    3. If the attempt was successful, tell authentication reducer about
    success and pass along user information
    4. otherwise, tell the authentication reducer about the failed attempt.
  */
  dispatch(apiRequest());
  try {
    const response = await axios.delete(RSVPS_URL, { email, eventID }, tokenConfig(getState));
    dispatch({
      type: DELETE_RSVP_SUCCESS,
      payload: response.data
    });
    return response;
  } catch (error) {
    dispatch({
      type: DELETE_RSVP_FAIL,
      error: error.message
    });
  }
}
