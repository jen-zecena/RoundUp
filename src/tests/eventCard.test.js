import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import EventCard from '../components/EventCard';
import { addRsvpAction } from '../actions/rsvpActions';
import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';



const mockStore = configureStore([]);

describe('My Connected React-Redux Component', () => {
    let store;
    let component;
   
    beforeEach(() => {
      store = mockStore({
        form: {
            rsvpForm: {
              registeredFields: {
                firstName: {
                  name: 'firstName',
                  type: 'Field',
                  count: 1
                },
                lastName: {
                  name: 'lastName',
                  type: 'Field',
                  count: 1
                },
                email: {
                  name: 'email',
                  type: 'Field',
                  count: 1
                },
                eID: {
                  name: 'eID',
                  type: 'Field',
                  count: 1
                }
              },
              syncErrors: {
                firstName: 'Required',
                lastName: 'Required',
                email: 'Required'
              }
            }
          },
          auth: {
            isLoading: false,
            isAuthenticated: false,
            user: null,
            token: null
          },
          events: {
            events: [
              {
                owner: 1,
                eID: 1,
                posterUrl: 'no poster',
                name: 'Test Event'
              },
              {
                owner: 1,
                eID: 44,
                posterUrl: 'https://roundupposters.s3.amazonaws.com/reunion2021_webbanners1500_x_866_2.jpg',
                name: 'Reunion'
              },
              {
                owner: 1,
                eID: 34,
                posterUrl: 'https://roundupposters.s3.amazonaws.com/poster7.jpg',
                name: 'Event'
              },
              {
                owner: 1,
                eID: 35,
                posterUrl: 'https://roundupposters.s3.amazonaws.com/poster1.jpg',
                name: '5C Event'
              },
              {
                owner: 1,
                eID: 32,
                posterUrl: 'https://roundupposters.s3.amazonaws.com/poster6.jpg',
                popularity: 0,
                eventTime: 1620709140000,
                name: 'Event 8',
                description: 'This is the test description for an event',
                location: 'cmc',
                status: 'active',
                tags: [
                  'Biology',
                  'Economics'
                ]
              }
            ],
            eventsWithTags: [],
            attendees: []
          },
          rsvps: {
            rsvps: []
          },
          subscriptions: {
            subscriptions: [],
            subscribers: []
          }
      });

      store.dispatch = jest.fn();

      component = renderer.create(
          <Provider store={store}>
              <EventCard/>
          </Provider>
      );
    });
   
    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
    
    it('component renders correctly', () => {
        expect(component).toContain(<Button variant="danger" > More Information </Button>).toBeTruthy();
        expect(component).toContain("Card.card").toBeTruthy();
        expect(component).toContain("Button.button").toBeTruthy();
        expect(component).toContain("Card.Title.title").toBeTruthy();
        expect(component).toContain("Card.Text.text").toBeTruthy();
        
    });

    // This one should work once rsvp works again
    // it('should dispatch an action on button click', () => {
    //     renderer.act(() => {
    //         component.root.findByType('button').props.onSubmit();
    //       });
       
    //       expect(store.dispatch).toHaveBeenCalledTimes(1);
    //       expect(store.dispatch).toHaveBeenCalledWith(
    //         addRsvpAction({ payload: "success" }) //update this payload with correct one 
    //       );
        
    // });
  });
//   https://jestjs.io/docs/expect this might contain other helpful examples of expect tests