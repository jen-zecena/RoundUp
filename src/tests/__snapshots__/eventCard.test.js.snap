// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="card card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
      <img
        className="card-img-top"
      />

  <div
    className="card-body"
  >
  <div
    className="tile card-title h5"
  >


  </div>
  <div 
    className="mb-2 text-muted card-subtitle h6"
  >


  </div>
  <p
    className="text card-text"
  />
  <a
    href=""
    onClick={[Function]}
  >
  <button
    className="button btn btn-danger"
    disabled={false}
    type="button"
  >

    More Information 
          
  </button>
 </a>
</div>
</div>
`;
